### PR TITLE
Switch to codecov from coveralls

### PR DESCRIPTION
**Pull request by waylan**
Fixes #581.

### CODE REVIEW

**Comment by waylan** [comment]
[codecov](https://codecov.io/gh/Python-Markdown/markdown)

Not sure why nothing is showing up in the checks below or on codecov.io.

**Comment by facelessuser** [comment]
Doesn't appear that there is a repo added: https://codecov.io/gh/Python-Markdown

**Comment by waylan** [comment]
Each time I  try to "add" the repo, it tells me the next step is to upload a report. Perhaps something is failing in Travis?

**Comment by facelessuser** [comment]
Maybe it doesn't kick in on pulls until it has a base from master.

**Comment by facelessuser** [comment]
Maybe because it's coming from waylan instead of a branch directly off PythonMarkdown....That's probably it.

**Comment by facelessuser** [comment]
Hey, we got something :).

File: .travis.yml [hunk 1]
```diff
@@ -12,19 +13,25 @@ matrix:
...
   apt:
     packages:
     - libtidy-0.99-0
     - aspell
     - aspell-en
+
 install:
   - pip install tox
-  - pip install coveralls
+  - pip install coverage<4.0
```
Comment (facelessuser) [inline .travis.yml]: Is there a reason for this (<4.0)?  I'm using coverage 4.0+ on all my projects without issue.

**Comment by waylan** [comment]
Yeah, I push a report from my local machine. Still failing from GitHub/Travis.

**Comment by facelessuser** [comment]
Hmm.  It was passing before your last commit (except it would then fail with no base to compare to).

https://codecov.io/gh/Python-Markdown/markdown/tree/908c2252e0ecb6056eb328f225887c78eecc8adf/markdown

**Comment by facelessuser** [comment]
I don't think you'll get a badge until it gets merged.

File: .travis.yml (hunk 1 above)
Comment (waylan) [reply .travis.yml]: See 20b17c29081. And the `.coverage` file format changed in 4.0. As the tox envs output the older format, we need the older version to upload as well.

**Comment by facelessuser** [comment]
Oh, I think I get what you mean...you pushed the report from your machine which is why we have a report...

**Comment by facelessuser** [comment]
Looks like that worked.  The "No report found to compare against" will go away once this is merged to master.  Though, why are we avoiding 4.0?

**Comment by waylan** [comment]
It works!

**Comment by waylan** [comment]
> why are we avoiding 4.0?

Because 4.0 dropped support for Python 3.2, which we still support (see 20b17c2). And the `.coverage` file format changed in 4.0. As the tox envs output the older format, we need the older version to upload as well.

I figured out that was the problem when I first tried to push a report from my local machine. Once I downgraded coverage, it worked.

File: .travis.yml (hunk 1 above)
Comment (facelessuser) [reply .travis.yml]: Interesting. I must not be covering 3.2... Or for some reason it doesn't cause an issue for me :/. Now I'm going to need to audit my repos.

**Comment by facelessuser** [comment]
Yup, I don't formally support 3.2 on any of my repos, so that is why I can get away with 4.0+.  Thanks for the explanation.

**Pull request merged**